src/App/Fossa/FossaAPIV1.hs

```diff
@@ -236,6 +271,87 @@ getLatestBuild apiOpts ProjectRevision{..} = fossaReq $ do
   response <- req GET (buildsEndpoint baseUrl orgId (Locator "custom" projectName (Just projectRevision))) NoReqBody jsonResponse baseOpts
   pure (responseBody response)
 
+---------- Archive build queueing. This Endpoint ensures that after an archive is uploaded, it is scanned.
+
+archiveBuildURL :: Url 'Https -> Url 'Https
+archiveBuildURL baseUrl = baseUrl /: "api" /: "components" /: "build"
+
+archiveBuildUpload ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  ApiOpts ->
+  ArchiveComponents ->
+  m (Maybe C.ByteString)
+archiveBuildUpload apiOpts archiveProjects = runEmpty $
+  fossaReqAllow401 $ do
+    (baseUrl, baseOpts) <- useApiOpts apiOpts
+
+    let opts = "dependency" =: True <> "rawLicenseScan" =: True
+        archiveBuildContext = "Queuing a build for an archive project"
+    -- The response appears to either be "Created" for new builds, or an error message for existing builds.
+    -- Making the actual return value of "Created" essentially worthless.
+    resp <- context archiveBuildContext $ req POST (archiveBuildURL baseUrl) (ReqBodyJson archiveProjects) bsResponse (baseOpts <> opts)
```

Comment:
[nit] I think it's easier to read `context` when it's inlined -- so I'd inline `archiveBuildContext` and put a newline after `$` for spacing

Reply:
That's fair `archiveBuildContext` is only used in one place as well so its not really necessary to do this.